src/config.rs

```diff
@@ -85,16 +85,33 @@ impl PersistentConfig {
             &url2digest_path,
             &app_dir(AppDataType::UserCache, &::APP_INFO, "manifests")?,
             &app_dir(AppDataType::UserCache, &::APP_INFO, "files")?,
-            status
-        )
+            status,
+        ).map(|bundle| Box::new(bundle) as _)
+    }
+
+    pub fn make_local_file_provider(
+        &self,
+        url: &str,
+        _status: &mut StatusBackend,
+    ) -> Result<Box<Bundle>> {
+        use std::path::Path;
+
+        let zb = ZipBundle::<File>::open(Path::new(url))?;
+
+        Ok(Box::new(zb) as _)
     }
 
     pub fn default_bundle(&self, status: &mut StatusBackend) -> Result<Box<Bundle>> {
         if self.default_bundles.len() != 1 {
             return Err(ErrorKind::Msg("exactly one default_bundle item must be specified (for now)".to_owned()).into());
         }
 
-        Ok(Box::new(self.make_cached_url_provider(&self.default_bundles[0].url, status)?))
+        let url = &self.default_bundles[0].url;
+        const FILE_SCHEME_PREFIX: &'static str = "file://";
+        if url.starts_with(FILE_SCHEME_PREFIX) {
+			return Ok(self.make_local_file_provider(&url[FILE_SCHEME_PREFIX.len()..], status)?);
```

Comment:
I am a little concerned about URL encoding here. A proper URL might contain a sequence like `%20` to encode a space, but this approach won't convert it as appropriate. I think the solution here is to parse the URL for real — Tectonic already depends on the `url` crate, so that should be used. Then you can check if the `scheme` is `file`, and use the handy [to_file_path](https://docs.rs/url/1.7.1/url/struct.Url.html#method.to_file_path) function.